hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making_config_map_pydantic.py

```diff
@@ -0,0 +1,278 @@
+        description="",
+        gt=0.0,
+        client_data=ClientFieldData(
+            prompt=lambda mi: (
+                "Enter conversion rate for taker base asset value to maker base asset value, e.g. "
+                "if maker base asset is USD and the taker is DAI, 1 DAI is valued at 1.25 USD, "
+                "the conversion rate is 1.25"
+            ),
+        ),
+    )
+    taker_to_maker_quote_conversion_rate: Decimal = Field(
+        default=Decimal("1.0"),
+        description="",
+        gt=0.0,
+        client_data=ClientFieldData(
+            prompt=lambda mi: (
+                "Enter conversion rate for taker quote asset value to maker quote asset value, e.g. "
+                "if maker quote asset is USD and the taker is DAI, 1 DAI is valued at 1.25 USD, "
+                "the conversion rate is 1.25"
+            ),
+        ),
+    )
+    slippage_buffer: Decimal = Field(
+        default=Decimal("5.0"),
+        description="",
+        ge=0.0,
+        le=100.0,
+        client_data=ClientFieldData(
+            prompt=lambda mi: (
+                "How much buffer do you want to add to the price to account for slippage for taker orders "
+                "Enter 1 to indicate 1%"
+            ),
+            prompt_on_new=True,
+        ),
+    )
+
+    # === prompts ===
+
+    @classmethod
+    def top_depth_tolerance_prompt(cls, model_instance: 'CrossExchangeMarketMakingConfigMap') -> str:
+        maker_market = model_instance.maker_market_trading_pair
+        base_asset, quote_asset = maker_market.split("-")
+        return f"What is your top depth tolerance? (in {base_asset})"
+
+    @classmethod
+    def order_amount_prompt(cls, model_instance: 'CrossExchangeMarketMakingConfigMap') -> str:
+        trading_pair = model_instance.maker_market_trading_pair
+        base_asset, quote_asset = trading_pair.split("-")
+        return f"What is the amount of {base_asset} per order?"
+
+    # === generic validations ===
+
+    @validator(
+        "adjust_order_enabled",
+        "active_order_canceling",
+        "use_oracle_conversion_rate",
+        pre=True,
+    )
+    def validate_bool(cls, v: str):
+        """Used for client-friendly error output."""
+        if isinstance(v, str):
+            ret = validate_bool(v)
+            if ret is not None:
+                raise ValueError(ret)
+        return v
+
+    @validator(
+        "min_profitability",
+        "order_amount",
+        "cancel_order_threshold",
+        "limit_order_min_expiration",
+        "top_depth_tolerance",
+        "anti_hysteresis_duration",
+        "order_size_taker_volume_factor",
+        "order_size_taker_balance_factor",
+        "order_size_portfolio_ratio_limit",
+        "taker_to_maker_base_conversion_rate",
+        "taker_to_maker_quote_conversion_rate",
+        "slippage_buffer",
+        pre=True,
+    )
+    def validate_decimal(cls, v: str, values: Dict, config: BaseModel.Config, field: Field):
```

Comment:
Again, neat idea! I still think we can simplify it further though.
Modifying a bit the code from the Slack message to handle the extra cases you're handling here:
```Python
def validate_decimal(v: str, field: Field):
    """Used for client-friendly error output."""
    field_info = field.field_info
    inclusive = field_info.ge is not None or field_info.le is not None
    min_value = field_info.gt if field_info.gt is not None else field_info.ge
    min_value = Decimal(min_value) if min_value is not None else min_value
    max_value = field_info.lt if field_info.lt is not None else field_info.le
    max_value = Decimal(max_value) if max_value is not None else max_value
    ret = validate_decimal(v, min_value, max_value, inclusive)
    if ret is not None:
        raise ValueError(ret)
    return v
```
And I propose to move this as a static method to the `BaseClientModel` class and simply use
```Python
    @validator(
        "min_profitability",
        "order_amount",
        "cancel_order_threshold",
        "limit_order_min_expiration",
        "top_depth_tolerance",
        "anti_hysteresis_duration",
        "order_size_taker_volume_factor",
        "order_size_taker_balance_factor",
        "order_size_portfolio_ratio_limit",
        "taker_to_maker_base_conversion_rate",
        "taker_to_maker_quote_conversion_rate",
        "slippage_buffer",
        pre=True,
    )
    def validate_decimal(cls, v: str, field: Field):
        """Used for client-friendly error output."""
        super().validate_decimal(v, field)
```

Reply:
✅